### ISSUE TITLE
Unable to build Hercules on Solaris 11

### ISSUE THREAD

**Issue by srorso** [hercules-390/hyperion#230]
Minor issues prevent building Hercules on Solaris.  Specifically:

- Solaris 11 sys/ioctl.h does not include sys/sockio.h unless the macro BSD_COMP is defined.  Other BSD systems  include sockio.h unconditionally.  Solution: perform a guarded include of sys/sockio.sys in hstdinc.h at either line 105 or 143 (hstdinc.h includes sys/ioctl.h in two places).
- Hercules source file `qeth.d` defines a function `static const char* sig2str( BYTE sig )` that Solaris 11 defines as  `extern int sig2str(int, char *)` in /usr/include/signal.h.
- Solaris 11 requires BSD-style naming of hifr_name in hifr.h at lines 112-116, not GNU/Linux naming.  
- The Solaris file command return `Hercules: ELF 32-bit LSB executable 80386 Version 1, ...`, a response not understood by runtest.  
- The runtest script is unable to find the redtest.rexx nor the test.conf files.  This remains under diagnosis.  

When the above issues are patched, Hercules builds cleanly (using CMake) and can IPL DOS/360.

**Comment by srorso** [comment]
The last item above,  "... runtest script is unable to find ...," is due to the Solaris Bourne Shell /bin/sh not supporting the -n option of the echo command.  The runtest script uses echo -n to determine whether command line arguments should be interpreted as positional or as labeled (-) flags.  See the following:
```
srorso@solaris113:~$ echo $SHELL
/usr/bin/bash
srorso@solaris113:~$ type echo
echo is a shell builtin
srorso@solaris113:~$ echo -n sample!
sample!srorso@solaris113:~$ /bin/sh
srorso@solaris113:~$ type echo
echo is a shell builtin
srorso@solaris113:~$ echo -n sample!                                                                              
-n sample!
srorso@solaris113:~$
```
The FreeBSD 11 Bourne Shell echo builtin _does_ support -n, _e.g.,_
```
login as: srorso
Using keyboard-interactive authentication.
Password for srorso@FreeBSD11:
Last login: Thu Jul 13 11:06:45 2017
FreeBSD 11.0-RELEASE-p9 (GENERIC) #0: Tue Apr 11 08:48:40 UTC 2017

Welcome to FreeBSD!
$ echo -n sample
sample$ type echo
echo is a shell builtin
$ echo $SHELL
/bin/sh
$
``` 
Removing -n from the echo command restores runtest functionality on Solaris.  Testing of -n removal is needed on other platforms.

**Comment by srorso** [comment]
Addressed in d05c651.
